specifications/EinsteinRiddle/Einstein.tla

```diff
@@ -51,17 +53,22 @@ Permutation(S) ==
 \* parameterization, the constants are replaced with constant-level
 \* operators.  As a side-effect, TLC evaluates them eagerly at startup, 
 \* and Apalache successfully determines their types.
-NATIONALITIES == Permutation({ "brit", "swede", "dane", "norwegian", "german" })
-DRINKS == Permutation({ "beer", "coffee", "mylk", "tea", "water" })
-COLORS == Permutation({ "red", "white", "blue", "green", "yellow" })
-PETS == Permutation({ "bird", "cat", "dog", "fish", "horse" })
-CIGARS == Permutation({ "blend", "bm", "dh", "pm", "prince" })
+NATIONALITIES == Permutation({ "brit_OF_NATIONALITY", "swede_OF_NATIONALITY", "dane_OF_NATIONALITY", "norwegian_OF_NATIONALITY", "german_OF_NATIONALITY" })
```

Comment:
```suggestion
BRIT == "brit_OF_NATIONALITY"
SWEDE == "swede_OF_NATIONALITY"
DANE == "dane_OF_NATIONALITY"
NORWEGIAN == "norwegian_OF_NATIONALITY"
GERMAN == "german_OF_NATIONALITY"
NATIONALITIES == Permutation({ BRIT, SWEDE, DANE, NORWEGIAN, GERMAN })
```

Here is my suggestion on how to avoid specifying the type all the time.